app/instances/shop.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+import datetime
+from typing import List
+
+from app.instances.customer import Customer
+
+
+@dataclass
+class Shop:
+
+    name: str
+    location: List[int]
+    products: dict
+
+    def serve_customer(self, customer: Customer) -> None:
+        total = 0
+
+        receipt = (
+            f"Date: {datetime.datetime.now().strftime('%d/%m/%Y %H:%M:%S')}"
+            f"\nThanks, {customer.name}, for your purchase!\n"
+            "You have bought: \n"
+        )
+
+        for product, quantity in customer.product_cart.items():
+            cost = self.products[product] * quantity
+
+            if isinstance(cost, float) and cost % 1 == 0:
+                cost = int(cost)
```

Comment:
Why do you change it to integer here?

Reply:
sometimes cost is equal 3.0, so if it's a case (cost is float and decimal part is 0) I change cost to int to have 3 printed out instead of 3.0